File: app/src/main/java/org/dhis2/usescases/splash/SplashContracts.java
package org.dhis2.usescases.splash;

import org.dhis2.usescases.general.AbstractActivityContracts;

import androidx.annotation.UiThread;
import io.reactivex.functions.Consumer;

public class SplashContracts {

    interface SplashView extends AbstractActivityContracts.View {

        Consumer<Integer> renderFlag();
    }

    interface SplashPresenter {
        void destroy();

        void init(SplashView view);

        @UiThread
        void isUserLoggedIn();

        @UiThread
        void navigateToLoginView();

        @UiThread
        void navigateToHomeView();
    }
}

File: app/src/main/java/org/dhis2/usescases/splash/SplashContracts.kt
package org.dhis2.usescases.splash

import androidx.annotation.UiThread
import org.dhis2.usescases.general.AbstractActivityContracts

class SplashContracts {

    interface View : AbstractActivityContracts.View {

        fun renderFlag(flagName: String)
    }

    interface Presenter {
        fun destroy()

        fun init(view: View)

        @UiThread
        fun isUserLoggedIn()

        @UiThread
        fun navigateTo(data: Class<*>)
    }
}
